eZ/Bundle/EzPublishCoreBundle/Command/NormalizeImagesPathsCommand.php

```diff
@@ -0,0 +1,127 @@
+
+        $this->imageGateway = $imageGateway;
+        $this->filePathNormalizer = $filePathNormalizer;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:normalize-image-paths')
+            ->setDescription('Normalizes stored paths for images.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Normalize images paths');
+
+        $io->writeln([
+            'Determining the number of images that require path normalization.',
+            'It may take a minute or two...',
+        ]);
+
+        $imagesData = $this->imageGateway->getAllImagesData();
+        $imagePathsToNormalize = [];
+        foreach ($imagesData as $imageData) {
+            $filePath = $imageData['filepath'];
+            $normalizedImagePath = $this->filePathNormalizer->normalizePath($imageData['filepath']);
+            if ($normalizedImagePath !== $filePath) {
+                $imagePathsToNormalize[] = [
+                    'fieldId' => (int) $imageData['contentobject_attribute_id'],
+                    'oldPath' => $filePath,
+                    'newPath' => $normalizedImagePath,
+                ];
+            }
+        }
+
+        $imagePathsToNormalizeCount = \count($imagePathsToNormalize);
+        $io->note(sprintf('Found: %d', $imagePathsToNormalizeCount));
+        if ($imagePathsToNormalizeCount === 0) {
+            $io->success('Nothing to do. Bye!');
+
+            return 0;
+        }
+
+        if (!$io->confirm('Do you want to continue?')) {
+            return 0;
+        }
+
+        $io->writeln('Normalizing images paths. Please wait...');
+        $io->progressStart($imagePathsToNormalizeCount);
+        foreach ($imagePathsToNormalize as $imagePathToNormalize) {
+            $this->updateImagePath(
+                $imagePathToNormalize['fieldId'],
+                $imagePathToNormalize['oldPath'],
+                $imagePathToNormalize['newPath']
+            );
+            $io->progressAdvance(1);
+        }
+        $io->progressFinish();
+        $io->success('Done!');
+
+        return 0;
+    }
+
+    private function updateImagePath(int $fieldId, string $oldPath, string $newPath): void
+    {
+        $oldPathInfo = pathinfo($oldPath);
+        $newPathInfo = pathinfo($newPath);
+        // In Image's XML, basename does not contain a file extension, and the filename does - pathinfo results are exactly the opposite.
+        $oldFileName = $oldPathInfo['basename'];
+        $newFilename = $newPathInfo['basename'];
+        $newBaseName = $newPathInfo['filename'];
+
+        $xmlsData = $this->imageGateway->getAllVersionsImageXmlForFieldId($fieldId);
+        foreach ($xmlsData as $xmlData) {
+            $dom = new \DOMDocument();
+            $dom->loadXml($xmlData['data_text']);
+
+            $ezimageTag = $dom->getElementsByTagName('ezimage')->item(0);
```

Comment:
Please avoid legacy naming where it can be avoided
```suggestion
            $imageTag = $dom->getElementsByTagName('ezimage')->item(0);
```